sdk/core/azure-core/src/http/policy_retry.cpp

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <http/policy.hpp>
+
+#include <algorithm>
+#include <cstdlib>
+#include <limits>
+#include <thread>
+
+using namespace Azure::Core::Http;
+
+namespace {
+decltype(RetryOptions::RetryDelay) ShouldRetry(
+    Response const& response,
+    RetryOptions const& retryOptions,
+    decltype(RetryOptions::MaxRetries) attempt)
+{
+  // Are we out of retry attempts?
+  if (attempt > retryOptions.MaxRetries)
+  {
+    return std::chrono::seconds(-1);
```

Comment:
Is this defined behavior?

Reply:
Again I don't think using a special value for a special meaning is a good approach. It creates ambiguity unnecessarily.
`ShouldRetry` should return `true` or `false`. The caller shouldn't be looking into the implementation to know what to expect as the return value.

Reply:
I agree for the public function, but this is helper. But I'll see if I can change it into boolean + out parameter.

Reply:
I meant it might be an undefined behavior. Undefined behavior is something we definitely need to avoid in our code.

Reply:
I updated to return via bool return type and out parameter.